core/sr-primitives/src/traits.rs

```diff
@@ -395,6 +395,8 @@ pub trait Hash: 'static + MaybeSerializeDeserialize + Debug + Clone + Eq + Parti
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 pub struct BlakeTwo256;
 
+// Search for Note [clippy::derive_hash_xor_eq] for an explanation of the line
```

Comment:
This technique is used by the Glasgow Haskell Compiler (GHC) developers, and I like it. :+1:

Reply:
Oh, you noticed! These notes are really helpful when I contribute to GHC, so that was the idea here as well.

However, I removed the `#[allow(clippy::...)]` flags as requested by one of your colleagues, so the comment does not make sense. I'll remove it.